app/controllers/orders_controller.rb

```diff
@@ -21,22 +21,49 @@ def show
   end
 
   def search
-    logger.tagged("#{current_user.urs_uid} #{controller_name}_controller") do
-      @orders = Echo::Orders.new(client: echo_client, echo_provider_token: echo_provider_token, guids: determine_order_guids).orders
+    Timeout.timeout(Rails.configuration.orders_timeout) do
+      logger.tagged("#{current_user.urs_uid} #{controller_name}_controller") do
+        request_start = Time.new
+        start = Time.new
+        guids = determine_order_guids
+        Rails.logger.info("Elapsed time to retrieve guids for #{guids.count} orders=#{Time.new - start}")
```

Comment:
It occurred to me it is doing this twice, it doesn't take very long, but needs to be moved up above line 25